fontforgeexe/charinfo.c

```diff
@@ -930,8 +930,31 @@ return( false );
     }
 
     /* If we get this far, then we didn't find any errors */
-    for ( pst = sc->possub; pst!=NULL; pst=pst->next )
-	pst->ticked = pst->type==pst_lcaret;
+    for ( pst = sc->possub; pst!=NULL; pst=pst->next ) {
+	pst->ticked = false;
+	if ( pst->type==pst_lcaret )
+	    lcpst = pst;
+    }
+
+    // There is no dialog datastructure for the ligature carets, so copy them here
+    // The entry will be modified in _CI_OK when needed
+    for ( pst = ci->sc->possub; pst!=NULL && pst->type!=pst_lcaret; pst=pst->next )
+	;
+    if ( pst!=NULL ) {
+	if ( lcpst==NULL ) {
+	    lcpst = chunkalloc(sizeof(PST));
+	    lcpst->type = pst_lcaret;
+	    lcpst->next = sc->possub;
+	    sc->possub = lcpst;
+	}
+	if ( lcpst->u.lcaret.carets!=NULL )
+	    free(lcpst->u.lcaret.carets);
```

Comment:
Is there a case in which extant carets would change via the dialog?

Reply:
Yes; definitely a 1->0 count transition and probably any count transition (the dialog lets you choose how many lcarets there are, just not where they are).